frontend/src/features/workspace/components/WorkspaceFormRow/RowActions/useRowAction.ts

```diff
@@ -79,14 +84,29 @@ export const useRowAction = (
     [formMeta, moveWorkspaceMutation],
   )
 
+  const handleRemoveFormFromWorkspaces = useCallback(async () => {
+    await removeFormFromWorkspacesMutation.mutateAsync({
+      formId: formMeta._id.toString(),
+    })
+  }, [formMeta, removeFormFromWorkspacesMutation])
+
+  const handleWorkspaceClick = useCallback(
+    (destWorkspace: Workspace, currFormWorkspace?: Workspace) => {
+      if (destWorkspace._id === currFormWorkspace?._id)
+        handleRemoveFormFromWorkspaces()
+      else handleMoveForm(destWorkspace._id.toString(), destWorkspace.title)
+    },
+    [handleMoveForm, handleRemoveFormFromWorkspaces],
+  )
+
   return {
     adminFormLink,
     previewFormLink,
     handleShareForm,
     handleDuplicateForm,
     handleCollaborators,
     handleDeleteForm,
-    handleMoveForm,
+    handleWorkspaceClick,
```

Comment:
Im thinking that we should export both `handleRemoveFormFromWorkspaces` and `handleMoveFormToWorkspace` instead of combining this action. I was having some difficulty making sense of this action, as it does not tell me what it is doing.

I think it's ok for be not-DRY for this function and have the callers `RowActionsDrawer`, and `RowActionsDropDown` duplicate some logic for better readability.

Reply:
Ahh I see, I guess it can be quite confusing. Cause was thinking technically both actions are from clicking the list of workspaces dropdown. We add if its not the current workspace of the form but remove when you click on the current workspace (the one with the tick). 

I think I sacrificed some readability for the sake of too much DRY-ness for both here and the `EmptyWorkspace` component above. Will refactor them to make it clearer! Thanks Ken!

Reply:
[88c43f0](https://github.com/opengovsg/FormSG/pull/6754/commits/88c43f0dcaa4ce41d9d89718fc2372a98ffc7099) 

Have also added some comments and renamed the method to make more readable!